File: examples/pybind11-project/python-src/pybind11_project/add_module.py
"""Example module that adds two integers in C++."""

import os
import typing

if not typing.TYPE_CHECKING and os.getenv("PYBIND11_PROJECT_PYTHON_DEBUG"):
    from ._add_module_d import *
    from ._add_module_d import __version__
else:
    from ._add_module import *
    from ._add_module import __version__

File: examples/nanobind-project/src/ext/add_module.cpp
/**
 * @file    Example C++ extension for Python using nanobind that adds two
 *          integers.
 */

#include <cstdint>
#include <nanobind/nanobind.h>
namespace nb = nanobind;
using namespace nb::literals;

int32_t add(int32_t a, int32_t b) {
    return a + b;
}

NB_MODULE(MODULE_NAME, m) {
    m.doc()               = "Module for adding integers";
    m.attr("__version__") = VERSION_INFO;
    m.def("add",              // function name in Python
          add,                // C++ function
          "a"_a, "b"_a,       // Argument names in Python
          "Adds two integers" // Docstring in Python
    );
}
